### ISSUE TITLE
Client is not entitled

### ISSUE THREAD

**Issue by VioletHill** [msintuneappsdk/ms-intune-app-sdk-ios#220]
I have integrated the Intune iOS SDK.  but show below error:

```
2021-12-17 15:02:39.053781+0800 IntuneDemo[10950:1122437] [unspecified] container_create_or_lookup_app_group_path_by_app_group_identifier: client is not entitled
2021-12-17 15:02:39.053937+0800 IntuneDemo[10950:1122437] [unspecified] container_create_or_lookup_app_group_path_by_app_group_identifier: client is not entitled
2021-12-17 15:02:39.054079+0800 IntuneDemo[10950:1122437] [unspecified] A NULL string is not a valid group container identifier.
2021-12-17 15:02:39.054097+0800 IntuneDemo[10950:1122437] [unspecified] container_create_or_lookup_app_group_path_by_app_group_identifier: client sent invalid parameters
Current file index 0
2021-12-17 15:02:40.715387+0800 IntuneDemo[10950:1122491] [connection] nw_socket_connect [C1:2] connectx(8 (guarded), [srcif=0, srcaddr=<NULL>, dstaddr=127.0.0.1:22], SAE_ASSOCID_ANY, 0, NULL, 0, NULL, SAE_CONNID_ANY) failed: [1: Operation not permitted]
2021-12-17 15:02:40.715680+0800 IntuneDemo[10950:1122491] [connection] nw_socket_connect [C1:2] connectx failed (fd 8) [1: Operation not permitted]
2021-12-17 15:02:40.715731+0800 IntuneDemo[10950:1122491] [] nw_socket_connect connectx failed [1: Operation not permitted]
2021-12-17 15:02:40.716120+0800 IntuneDemo[10950:1122491] Connection 1: received failure notification
````

the first error is "container_create_or_lookup_app_group_path_by_app_group_identifier: client is not entitled"

I sure that I have followed the [doc](https://docs.microsoft.com/en-us/mem/intune/developer/app-sdk-ios) and Enable keychain sharing

<img width="712" alt="image" src="https://user-images.githubusercontent.com/2719779/146503855-712c27d0-9824-44b2-a0a5-6f75b9fc58a7.png">

Any problem? is that because there's no permission to access the keychain and how do I resolve this issue?

Env:
Xcode : iOS 13.1
iOS: iOS 15.1.1

**Comment by VioletHill** [comment]
close this issue by resolving using correct 'CFBundleURLSchemes'

**Comment by surajindoqubix** [comment]
Hi @VioletHill,
Getting same issue. How did you resolved this?

**Comment by VioletHill** [comment]
> Hi @VioletHill,
> Getting same issue. How did you resolved this?

hi @surajindoqubix. check your CFBundleURLSchemes. if your URLSchemes are not right. it will not work. I met this issue first time because my CFBundleURLSchemes is not right

**Comment by surajindoqubix** [comment]
Hi @VioletHill ,

I also thought so, but have verified same multiple times from doc also but still getting same error in console. 

<img width="748" alt="Screenshot 2022-05-16 at 9 58 38 PM" src="https://user-images.githubusercontent.com/104090623/168640065-c4d6c2de-7107-48e8-bd10-dce070834d31.png">
<img width="917" alt="Screenshot 2022-05-16 at 9 58 29 PM" src="https://user-images.githubusercontent.com/104090623/168640073-bb8e1cf1-e6f1-4349-a6f0-e06f5005b69d.png">


2022-05-16 21:57:25.091780+0530 CareMember[346:5855] [unspecified] container_create_or_lookup_app_group_path_by_app_group_identifier: client is not entitled
2022-05-16 21:57:25.092735+0530 CareMember[346:5855] [unspecified] container_create_or_lookup_app_group_path_by_app_group_identifier: client is not entitled
2022-05-16 21:57:25.094800+0530 CareMember[346:5855] [unspecified] A NULL string is not a valid group container identifier.
2022-05-16 21:57:25.094883+0530 CareMember[346:5855] [unspecified] container_create_or_lookup_app_group_path_by_app_group_identifier: client sent invalid parameters
2022-05-16 21:57:25.301386+0530 CareMember[346:5855] [unspecified] container_create_or_lookup_app_group_path_by_app_group_identifier: client is not entitled
2022-05-16 21:57:25.301658+0530 CareMember[346:5855] [unspecified] container_create_or_lookup_app_group_path_by_app_group_identifier: client is not entitled
2022-05-16 21:57:25.301691+0530 CareMember[346:5855] [unspecified] A NULL string is not a valid group container identifier.
2022-05-16 21:57:25.301707+0530 CareMember[346:5855] [unspecified] container_create_or_lookup_app_group_path_by_app_group_identifier: client sent invalid parameters
2022-05-16 21:57:25.454735+0530 CareMember[346:5855] [unspecified] container_create_or_lookup_app_group_path_by_app_group_identifier: client is not entitled
2022-05-16 21:57:25.455201+0530 CareMember[346:5855] [unspecified] container_create_or_lookup_app_group_path_by_app_group_identifier: client is not entitled
2022-05-16 21:57:25.455252+0530 CareMember[346:5855] [unspecified] A NULL string is not a valid group container identifier.
2022-05-16 21:57:25.455277+0530 CareMember[346:5855] [unspecified] container_create_or_lookup_app_group_path_by_app_group_identifier: client sent invalid parameters

**Comment by VioletHill** [comment]
looks like your CFBundleURLSchemes are not right.
it should be 
<key>CFBundleURLSchemes</key>
			<array>
				<string>com.intunedemo.mobile</string>
				<string>com.intunedemo.mobile-intunemam</string>
				<string>msauth.intunedemo.mobile</string>
			</array>

**Comment by neiljac1** [comment]
Hi @surajindoqubix 

Please check to make sure the AppGroupIdentifiers listed under IntuneMAMSettings in your Info.plist match your application's com.apple.security.application-groups entitlements.

Thanks,
Neil

**Comment by neiljac1** [comment]
Hi @AceChen1 . After further investigation, we determined these error messages are generated by SDK code making a call to NSFileManager containerURLForSecurityApplicationGroupIdentifier on an invalid app group contain name. This issue will be fixed in a future release.

Thank you,
Neil

**Comment by AceChen1** [comment]
> Hi @AceChen1 . After further investigation, we determined these error messages are generated by SDK code making a call to NSFileManager containerURLForSecurityApplicationGroupIdentifier on an invalid app group contain name. This issue will be fixed in a future release.
> 
> Thank you, Neil

@neiljac1  Hi Neil, Thanks your help, then let me wait your guys to release the new version, then try try.

**Comment by AceChen1** [comment]
@neiljac1  hello, looks like my app still crash when upgrade the sdk verion to 16.0.5, may i know the issue had been fixed on the 16.0.5 or not.

**Comment by symfony-21-teen-age6** [comment]
I have the insight now https://stackdev.space/questions/organizing-xcode-projects-into-different-categories
